File: CPP_Algorithms/Math/gcd.go
package main

import "fmt"

func FindGCD(a int, b int) int {
	if a == 0 {
		return b
	}
	return FindGCD(a-b, b)
}

func main() {
	var a, b int
	fmt.Println("Enter the two numbers to find GCD")
	fmt.Scan(&a, &b)
	res := FindGCD(a, b)
	fmt.Printf("The GCD of two numbers is %d", res)

}

File: C++/Basic/GCD.cpp
#include<iostream>
using namespace std;
int gcd (int a, int b ){
       if (a==0 ){
            return b;
        }
        if (b==0 ){
            return a;
        }
    while (a!=b){
     
        if (a>b){
            a=a-b;
        }
        else {
            b=b-a;
        }
    }
    return a;
}
int main(){
    int n1,n2;
    cout<<"Enter numbers :-";
    cin>>n1>>n2;
    int ans=gcd(n1, n2);
    cout<<"GCD is ->"<<ans;
    return 0;
}